finality-aleph/src/party/manager/mod.rs

```diff
@@ -350,11 +353,14 @@ where
         };
 
         let last_block_of_previous_session = session_boundaries.first_block().saturating_sub(1);
-        let last_block_of_previous_session_hash = self
-            .client
-            .block_hash(last_block_of_previous_session)
-            .expect("Previous session ended, the block should be present")
-            .expect("Previous session ended, we should have the hash.");
+        let last_block_of_previous_session_header = self
+            .header_backend
+            .header_of_finalized_at(last_block_of_previous_session)
```

Comment:
Double-checking - last block of previous session should be already finalized, right?

Reply:
yep

Reply:
Yup!